src/lib/memoise.ts

```diff
@@ -0,0 +1,16 @@
+export const memoise = <T>(fn: () => Promise<T>): [() => void, () => Promise<T>] => {
```

Comment:
Thinking about the interface to this - I wonder if it's the right call to tie this to async/Promise based return values. We could also make it more generic and push the async part (basically just the `await`) to the caller. What do you think? Or otherwise maybe we could rename to make it's purpose clearer (`memoiseAsync`)?

Reply:
yeah I think renaming to `memoiseAsync` is a good shout. I *think* I need await in the body to do things right/can't make this generic across async and non-async unfortunately.

Reply:
:+1: That sounds valid. Yeah since we want to maintain the functionality of not memoizing a failed promise then we need to keep this async aware.